### DISCUSSION TITLE
how to add a bed track

### DISCUSSION THREAD

**Discussion by qwzhang0601** [GMOD/jbrowse-components#1449] [Q&A]
I am trying to add a track for my bed file, I used the command "flatfile-to-json.pl --bed ./data.input/bed/i4_i5.bed --tracklabel i4i5 --out ./data.apollo", but get following errors. I can prepare gff track successfully. Not clear what's wrong. Do you have any suggestions? thanks. 

#errors
Can't locate Bio/FeatureIO.pm in @INC (you may need to install the Bio::FeatureIO module) (@INC contains: /Users/qwzhang/Documents/tools/Apollo/jbrowse/bin/../src/perl5 /Users/qwzhang/Documents/tools/vcftools_0.1.13/perl /Users/qwzhang/Library/ActivePerl-5.26/lib /usr/local/ActivePerl-5.26/site/lib /usr/local/ActivePerl-5.26/lib) at /Users/qwzhang/Documents/tools/Apollo/jbrowse/bin/../src/perl5/Bio/JBrowse/Cmd/FlatFileToJson.pm line 158.
 at /Users/qwzhang/Documents/tools/Apollo/jbrowse/bin/../src/perl5/Bio/JBrowse/ConfigurationManager.pm line 7.
	Bio::JBrowse::ConfigurationManager::__ANON__("Can't locate Bio/FeatureIO.pm in \@INC (you may need to instal"...) called at /Users/qwzhang/Documents/tools/Apollo/jbrowse/bin/../src/perl5/Bio/JBrowse/Cmd/FlatFileToJson.pm line 158
	Bio::JBrowse::Cmd::FlatFileToJson::make_bed_stream(Bio::JBrowse::Cmd::FlatFileToJson=HASH(0x7fa77d8030a0)) called at /Users/qwzhang/Documents/tools/Apollo/jbrowse/bin/../src/perl5/Bio/JBrowse/Cmd/FlatFileToJson.pm line 118
	Bio::JBrowse::Cmd::FlatFileToJson::run(Bio::JBrowse::Cmd::FlatFileToJson=HASH(0x7fa77d8030a0)) called at /Users/qwzhang/Documents/tools/Apollo/jbrowse/bin/flatfile-to-json.pl line 9

#My bed file
Contig25        10847917        10847543        i4.1    100     -
Contig25        10850493        10850351        i4.2    100     -
Contig25        10841461        10840918        i5.1    99      -
Contig25        10850686        10850351        i5.2    99      -
Contig25        10854061        10853736        i5.3    97      -
Contig25        10854358        10854150        i5.4    100     -
Contig25        10847018        10846971        i5.5    100     -
Contig25        10847917        10847868        i5.6    98      -

**Comment by cmdcolin** [comment]
@qwzhang0601 sorry for the delay, this is the jbrowse 2 forum technically

this appears to be somewhat an Apollo issue, since it appears to suggest that at the least jbrowse's dependencies were not installed correctly by the apollo installation


I might recommend to download jbrowse by itself, run setup.sh in jbrowse, so you can use jbrowse tools. That way you are not bound by possible issues with Apollo's installation of the jbrowse tools.
